Python/Product/Analysis/Values/FunctionInfo.cs

```diff
@@ -115,9 +115,11 @@ internal class FunctionInfo : AnalysisValue, IReferenceableContainer, IHasRichDe
                     calledUnit.Enqueue();
                 }
 
-                res.Split(v => v.IsResolvable(), out _, out var nonLazy);
-                res = DoCall(node, unit, calledUnit, callArgs);
-                res = res.Union(nonLazy);
+                if (calledUnit != null) {
```

Comment:
If we don't have a specific called unit here, we should get res from `_analysisUnit.ReturnValue.Types` (which should have the union of all possible return types for this function).

Reply:
It's probably easiest to make `DoCall` handle a null calledUnit. We can also assert `calledUnit != null || callingUnit.ForEval` in there, since we should never end up in a situation where we are still analyzing but get a null `calledUnit`.

Reply:
I got null ref here since condition above does not guarantee it to be non-null (i.e. .ForEval case)

Reply:
Perhaps it is better to add `|| calledUnit == null` to the `if` above